### PR TITLE
Imu: update calls to use sdf::Errors output

### PR DESCRIPTION
**Pull request by marcoag**
Signed-off-by: Marco A. Gutierrez <marco@openrobotics.org>

# 🎉 New feature

Work towards https://github.com/gazebosim/sdformat/issues/820.

Depends on https://github.com/gazebosim/sdformat/pull/1151, which depends on: https://github.com/gazebosim/sdformat/pull/1141.

## Summary
Adds missing `Errors` structure parameters in a few methods of the `Imu` class.

## Test it

Using the `Imu` class should report all errors through `sdf::Errors` if the `errors` parameter is used, it should print them otherwise.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/Imu.cc [hunk 1]
```diff
@@ -96,96 +97,98 @@ Errors Imu::Load(ElementPtr _sdf)
...
-    sdf::ElementPtr elem = _sdf->GetElement("linear_acceleration");
+    sdf::ElementPtr elem = _sdf->GetElement("linear_acceleration", errors);
     if (elem->HasElement("x"))
     {
-      if (elem->GetElement("x")->HasElement("noise"))
+      if (elem->GetElement("x", errors)->HasElement("noise"))
       {
         this->dataPtr->linearAccelXNoise.Load(
-            elem->GetElement("x")->GetElement("noise"));
+            elem->GetElement("x", errors)->GetElement("noise", errors));
```
Comment (aaronchongth) [inline src/Imu.cc:106]: We'll probably want to grab the errors from `Noise::Load` as well, perhaps something like,

```
sdf::Errors noiseErrors = this->dataPtr->linearAccelXNoise.Load(
```

before adding `noiseErrors` into `errors` as well. This probably applies to all the other `Noise::Load` calls below too

File: include/sdf/Imu.hh [hunk 2]
```diff
@@ -260,6 +260,14 @@ namespace sdf
     /// \return SDF element pointer with updated sensor values.
     public: sdf::ElementPtr ToElement() const;
 
+    /// \brief Create and return an SDF element filled with data from this
+    /// imu sensor.
+    /// Note that parameter passing functionality is not captured with this
+    /// function.
+    /// \param[out] _errors Vector of errors.
+    /// \return SDF element pointer with updated sensor values.
+    public: sdf::ElementPtr ToElement(sdf::Errors &_errors) const;
```
Comment (aaronchongth) [inline include/sdf/Imu.hh:269]: We should probably add a couple of unit tests for this overload too

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Thanks! I just left a few comments, let me know what you think

File: src/Imu.cc (hunk 1 above)
Comment (marcoag) [reply src/Imu.cc:106]: Good catch! `Load` errors should be grabbed wherever the call happens.

**Review by aaronchongth — APPROVED** [review]
LGTM!

**Comment by marcoag** [comment]
Added a quick test for the `ToElement` and `Load` errors output.

**Pull request merged**